internal/admission-webhook-server/admission-webhook-server.go

```diff
@@ -0,0 +1,146 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package admissionwebhookserver contains code for admission webhook.
+package admissionwebhookserver
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	v1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// ServeMutateServiceAccount is a service to mutate default service account.
+func ServeMutateServiceAccount(w http.ResponseWriter, r *http.Request) {
+	serve(w, r, mutateServiceAccount)
+}
+
+type patchOperation struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value,omitempty"`
+}
+
+type admitFunc func(v1.AdmissionReview) *v1.AdmissionResponse
+
+func toAdmissionResponse(err error) *v1.AdmissionResponse {
+	return &v1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
+func validateRequest(r *http.Request) ([]byte, error) {
+	var body []byte
+
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	if len(body) == 0 {
+		return nil, errors.New("empty body")
+	}
+
+	// Verify the content type is accurate.
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		return nil, errors.New("Content-Type=" + contentType + ", expect application/json")
+	}
+
+	return body, nil
+}
+
+func serve(w http.ResponseWriter, r *http.Request, admit admitFunc) {
+	body, err := validateRequest(r)
+	if err != nil {
+		glog.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	// The AdmissionReview that was sent to the webhook.
+	requestedAdmissionReview := v1.AdmissionReview{}
+
+	// The AdmissionReview that will be returned.
+	responseAdmissionReview := v1.AdmissionReview{}
+
+	deserializer := scheme.Codecs.UniversalDeserializer()
+
+	if _, _, err := deserializer.Decode(body, nil, &requestedAdmissionReview); err != nil {
+		glog.Error(err)
+		responseAdmissionReview.Response = toAdmissionResponse(err)
+	} else {
+		responseAdmissionReview.Response = admit(requestedAdmissionReview)
+	}
+
+	responseAdmissionReview.Response.UID = requestedAdmissionReview.Request.UID
+	responseAdmissionReview.APIVersion = "admission.k8s.io/v1"
+	responseAdmissionReview.Kind = "AdmissionReview"
+
+	respBytes, err := json.Marshal(responseAdmissionReview)
+	if err != nil {
+		glog.Error(err)
+	}
+
+	if _, err := w.Write(respBytes); err != nil {
+		glog.Error(err)
+	}
+}
+
+func mutateServiceAccount(ar v1.AdmissionReview) *v1.AdmissionResponse {
+	req := ar.Request
+
+	glog.Infof("AdmissionReview for Kind=%v, Namespace=%v Name=%v UID=%v patchOperation=%v UserInfo=%v",
+		req.Kind, req.Namespace, req.Name, req.UID, req.Operation, req.UserInfo)
+
+	if req.Kind.Kind == "ServiceAccount" && req.Name == "default" {
```

Comment:
I'd revert the logic to keep the happy path close to left side:
```suggestion
	if req.Kind.Kind != "ServiceAccount" || req.Name != "default" {
		glog.Infof("Skipping mutation for Kind=%v Name=%v", req.Kind, req.Name)

	        return &v1.AdmissionResponse{
	        	  Allowed: true,
	        }
	 }
```